components/post_view/reaction/index.js

```diff
@@ -33,8 +39,8 @@ function makeMapStateToProps() {
         }
         const channel = getChannel(state, {id: ownProps.post.channel_id}) || {};
         const teamId = channel.team_id;
-        const canAddReaction = haveIChannelPermission(state, {team: teamId, channel: ownProps.post.channel_id, permission: Permissions.ADD_REACTION});
-        const canRemoveReaction = haveIChannelPermission(state, {team: teamId, channel: ownProps.post.channel_id, permission: Permissions.REMOVE_REACTION});
+        const canAddReaction = haveIChannelPermission(state, {team: teamId, channel: ownProps.post.channel_id, permission: Permissions.ADD_REACTION}) && (!(channel.Name === Constants.DEFAULT_CHANNEL && config.ExperimentalTownSquareIsReadOnly === 'true' && license.IsLicensed === 'true') || me.roles.includes('system_admin'));
```

Comment:
Can we move the different conditions to variables to make it easier to read. No is really hard to understand the logic behind